### PR TITLE
Credentials in store

### PR DESCRIPTION
**Pull request by clguimanMSFT**
* Added AzureCredentials class - handles azure service principals
* Refactored the code to use AzureCredentials.ServicePrincipal instead of each individual credential parameter (subscriptionId, clientId, secret, token, URL)
* Agent templates use password credentials (Matt's work)

### CODE REVIEW

File: src/main/java/com/microsoft/azure/AzureVMManagementServiceDelegate.java [hunk 1]
```diff
@@ -1069,13 +1049,6 @@ private static String getLocationName(String location) {
                 return errors;
             }
 
-            //verify password
-            validationResult = verifyAdminPassword(adminPassword);
-            addValidationResultIfFailed(validationResult, errors);
-            if (returnOnSingleError && errors.size() > 0) {
-                return errors;
-            }
-
```
Comment (mmitche) [inline src/main/java/com/microsoft/azure/AzureVMManagementServiceDelegate.java]: This should probably be added back in.  I removed this for bringup but IIRC the deployment will fail if the password is too weak. It just needs to look up the password first.

File: src/main/resources/com/microsoft/azure/util/AzureCredentials/credentials.jelly [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    <f:entry title="${%OAuth 2.0 Token Endpoint}" field="oauth2TokenEndpoint" help="/plugin/azure-vm-agents-plugin/help-oauth2TokenEndpoint.html">
+      <f:password />
+    </f:entry>
+    <f:entry title="${%Management Service URL}" field="serviceManagementURL" help="/plugin/azure-vm-agents-plugin/help-serviceManagementURL.html">
+      <f:textbox default="${descriptor.getDefaultserviceManagementURL()}"/>
+    </f:entry>
+    <st:include page="id-and-description" class="${descriptor.clazz}"/>
+    <f:validateButton title="${%Verify Configuration}" progress="${%Verifying}" method="verifyConfiguration" 
+          with="subscriptionId,clientId,clientSecret,oauth2TokenEndpoint,serviceManagementURL" />
+</j:jelly>
```
Comment (mmitche) [inline src/main/resources/com/microsoft/azure/util/AzureCredentials/credentials.jelly]: What's the purpose of this file?  I thought we're now loading all the credentials from the credential store?

Comment (clguimanMSFT) [inline src/main/resources/com/microsoft/azure/util/AzureCredentials/credentials.jelly]: That's the UI for Azure Credentials

File: src/main/java/com/microsoft/azure/AzureVMManagementServiceDelegate.java (hunk 1 above)
Comment (clguimanMSFT) [inline src/main/java/com/microsoft/azure/AzureVMManagementServiceDelegate.java]: fixed

**Comment by arroyc** [comment]
looks good, merging ...

**Pull request merged**